srcpkgs/tbb/template

```diff
@@ -1,76 +1,43 @@
 # Template file for 'tbb'
 pkgname=tbb
-version=2020.U3
+version=2021.5.0
 revision=1
-# version rewrite: 2017_U7 (upstream) => 2017.7 (xbps)
-_ver=${version/./_}
-wrksrc="oneTBB-${_ver}"
-build_style=gnu-makefile
-hostmakedepends="cmake"
+wrksrc="oneTBB-${version}"
+build_style=cmake
+makedepends="libgomp-devel libhwloc-devel"
 short_desc="Intel Threading Building Blocks"
 maintainer="Andrea Brancaleoni <abc@pompel.me>"
 license="Apache-2.0"
 homepage="https://www.threadingbuildingblocks.org"
-distfiles="https://github.com/oneapi-src/oneTBB/archive/${_ver}.tar.gz"
-checksum=2103cc6238c935664f87680618f6684d57501d4a2fa8ea8f6c97ad6ff7dc722a
+changelog="https://raw.githubusercontent.com/oneapi-src/oneTBB/master/RELEASE_NOTES.md"
+distfiles="https://github.com/oneapi-src/oneTBB/archive/refs/tags/v${version}.tar.gz"
+checksum=e5b57537c741400cf6134b428fc1689a649d7d38d9bb9c1b6d64f092ea28178a
 
-case "$XBPS_TARGET_MACHINE" in
-	x86_64*)
-		make_build_args="arch=intel64"
-		;;
-	i686*)
-		make_build_args="arch=ia32"
-		;;
-	armv7l*)
-		make_build_args="arch=armv7"
-		;;
-	aarch64*)
-		make_build_args="arch=arm64"
-		;;
-	ppc64*)
-		make_build_args="arch=ppc64"
-		;;
-	ppc*)
-		make_build_args="arch=ppc32"
-		;;
-	*)
-		# make sure to fall back instead of letting host uname control it
-		make_build_args="arch=generic"
-		;;
-esac
+if [ "$XBPS_TARGET_NO_ATOMIC8" ]; then
+	makedepends+=" libatomic-devel"
+	CFLAGS+=" -latomic"
+fi
 
-post_extract() {
-	sed -i 's|CPLUS = g..|CPLUS = $(CXX)|g' build/linux.gcc.inc
-	sed -i 's|CONLY = gcc|CONLY = $(CC)|g' build/linux.gcc.inc
+if [ "$XBPS_TARGET_LIBC" = "musl" ]; then
+	makedepends+=" libucontext-devel"
+fi
```

Comment:
This has the unfortunate side effect of making it only really work on x86 and aarch64 :/

Idk how many users on ARM we have, though. And well, there's not much we can do.

Reply:
It's not working here either, so we don't have to worry about that yet. Are there any ways to get around the `undefined reference to 'swapcontext', 'getcontext', 'makecontext'` errors?